osu.Game/Overlays/FirstRunSetupOverlay.cs

```diff
@@ -0,0 +1,397 @@
+        {
+            base.LoadComplete();
+
+            config.BindWith(OsuSetting.ShowFirstRunSetup, showFirstRunSetup);
+
+            if (showFirstRunSetup.Value) Show();
+        }
+
+        public override bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
+        {
+            if (!e.Repeat)
+            {
+                switch (e.Action)
+                {
+                    case GlobalAction.Select:
+                        NextButton.TriggerClick();
+                        return true;
+
+                    case GlobalAction.Back:
+                        if (BackButton.Enabled.Value)
+                        {
+                            BackButton.TriggerClick();
+                            return true;
+                        }
+
+                        // If back button is disabled, we are at the first step.
+                        // The base call will handle dismissal of the overlay.
+                        break;
+                }
+            }
+
+            return base.OnPressed(e);
+        }
+
+        public override void Show()
+        {
+            // if we are valid for display, only do so after reaching the main menu.
+            performer.PerformFromScreen(screen =>
+            {
+                MainMenu menu = (MainMenu)screen;
+
+                // Eventually I'd like to replace this with a better method that doesn't access the screen.
+                // Either this dialog would be converted to its own screen, or at very least be "hosted" by a screen pushed to the main menu.
+                // Alternatively, another method of disabling notifications could be added to `INotificationOverlay`.
+                if (menu != null)
+                {
+                    overlayActivationMode = menu.OverlayActivationMode.GetBoundCopy();
+                    overlayActivationMode.Value = OverlayActivation.UserTriggered;
+                }
+
+                base.Show();
+            }, new[] { typeof(MainMenu) });
+        }
+
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            this.ScaleTo(scale_when_hidden)
+                .ScaleTo(1, 400, Easing.OutElasticHalf);
+
+            this.FadeIn(400, Easing.OutQuint);
+
+            if (currentStepIndex == null)
+                showFirstStep();
+        }
+
+        protected override void PopOut()
+        {
+            if (overlayActivationMode != null)
+            {
+                // If this is non-null we are guaranteed to have come from the main menu.
+                overlayActivationMode.Value = OverlayActivation.All;
+                overlayActivationMode = null;
+            }
+
+            if (currentStepIndex != null)
+            {
+                notificationOverlay.Post(new SimpleNotification
+                {
+                    Text = FirstRunSetupOverlayStrings.ClickToResumeFirstRunSetupAtAnyPoint,
+                    Icon = FontAwesome.Solid.Horse,
```

Comment:
Out of curiosity more than anything, does this horse... signify anything? Because it looks a lot like this might have been a placeholder for the time being, so I'm not sure if it should stay :sweat_smile:

Reply:
About as placeholder as the pile of poo that was on the disclaimer screen. Am very open to suggestion before merge.

Reply:
Bit of a difficult one to pick a sensible icon for... Maybe a "redo" one?

![image](https://user-images.githubusercontent.com/20418176/164318938-6f82b7a1-8851-4948-b835-2268184a9c17.png)

```suggestion
                    Icon = FontAwesome.Solid.Redo,
```

Reply:
Sure, let's go with that for now. RIP horse.